src/main/java/com/zaxxer/nuprocess/linux/EpollEvent.java

```diff
@@ -79,7 +83,7 @@ public static class EpollEventPrototype extends Structure
       public int events;
       public EpollData data;
 
-      EpollEventPrototype() {
+      public EpollEventPrototype() {
```

Comment:
Why did this need to be made `public`? (That said, though, in light of other comments I've added, I think you can just revert this file entirely.)

Reply:
It is for using the JNA Struct API. It requires public constructor.

I am scratching my head a bit, why are there both the JNA Struct (EpollEventPrototype) and the POJO EpollEvent for the same thing? Hm.

I don't mind the extra few java objects. But we can do better.

Reply:
Having two objects is part of "doing better". JNA `Structure`s are expensive to allocate, in a number of ways. The approach taken here is to instantiate the structure _once_ and use that to understand the memory layout. Then, going forward, `EpollEvent` is a simple POJO which allocates a simple `Memory` region of the right size. `EpollEvent` has no fields as such; it just writes into the memory region at the right locations for the fields the structure had.

I linked to the commit where that optimization was made, and in pull request #80 there are comments which show memory churn before and after that change--showing it's a pretty significant win.